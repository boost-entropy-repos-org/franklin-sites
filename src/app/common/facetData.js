const facetData = [
  {
    label: 'Facet 1',
    name: 'facet_1',
    allowMultipleSelection: false,
    values: [
      {
        label: 'Value 1',
        value: 'value_1',
        count: 76,
      },
      {
        label: 'Value 2',
        value: 'value_2',
        count: 79,
      },
    ],
  },
  {
    label: 'Facet 2',
    name: 'facet_2',
    allowMultipleSelection: true,
    values: [
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: undefined,
        value: 'value_2',
        count: 794,
      },
    ],
  },
  {
    label: 'Facet not shown',
    name: 'facet_not',
  },
  {
    label: 'Long facet shown',
    name: 'long_facet',
    values: [
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
      {
        label: 'Value 1',
        value: 'value_1',
        count: 764879,
      },
    ],
  },
];

export default facetData;
