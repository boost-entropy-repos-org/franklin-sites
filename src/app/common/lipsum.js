import { loremIpsum } from 'lorem-ipsum';
import v1 from 'uuid';

export const getLipsumData = ({ numberDataPoints, keys, idKey }) => Array(numberDataPoints)
  .fill(null)
  .map(() => {
    const dataPoint = { [idKey]: v1() };
    keys.forEach((key) => {
      dataPoint[key] = {
        value: loremIpsum({
          sentenceLowerBound: 2,
          sentenceUpperBound: 30,
        }),
      };
    });
    return dataPoint;
  });

export default getLipsumData;
