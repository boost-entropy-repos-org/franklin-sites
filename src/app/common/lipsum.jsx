import { loremIpsum } from 'lorem-ipsum';
import v1 from 'uuid';

export const getLipsumSentence = () =>
  loremIpsum({
    sentenceLowerBound: 2,
    sentenceUpperBound: 30,
  });

export const fillArray = (numberElements, func) =>
  Array(numberElements)
    .fill(null)
    .map(func);

export const getLipsumObjectArray = ({ numberElements, keys, idKey }) =>
  fillArray(numberElements, () => {
    const dataPoint = { [idKey]: v1() };
    keys.forEach((key) => {
      const text = getLipsumSentence();
      dataPoint[key] = text;
    });
    return dataPoint;
  });

export default getLipsumObjectArray;
