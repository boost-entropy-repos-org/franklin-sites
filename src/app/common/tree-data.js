export const treeData = [{
  label: 'Item 1',
  term: 'item_1',
  items: [{
    label: 'Item 1 a',
    term: 'item_1_a',
  }, {
    label: 'Item 1 b',
    term: 'item_1_b',
    items: [{
      label: 'Item 1 b A',
      term: 'item_1_b_A',
    }, {
      label: 'Item 1 b B',
      term: 'item_1_b_B',
    }]
  }]
}, {
  label: 'Some Item 2',
  term: 'item_2'
}];

function getPathLabel(path, sep = ' / ') {
  return path.map(item => item.label).join(sep);
}

function flattenTreeData(treeData, path = []) {
  let flattened = [];
  for (const node of treeData) {
    let nodePath = [...path, node]
    flattened = [...flattened, {
      'label': getPathLabel(nodePath),
      'value': nodePath
    }];
    if (node.items) {
      const moreFlattened = flattenTreeData(node.items, nodePath);
      if (moreFlattened.length) {
        flattened = [...flattened, ...moreFlattened];
      }
    }
  }
  return flattened;
}

export const flatTreeData = flattenTreeData(treeData);
