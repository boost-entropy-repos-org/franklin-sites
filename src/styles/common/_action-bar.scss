.action-bar {
  display: flex;
  justify-content: flex-end;
  .button {
    margin: 0 0.5em;
  }
}
