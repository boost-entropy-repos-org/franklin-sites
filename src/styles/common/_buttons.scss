@import '../settings';
@import '../colours';

.button {
  svg {
    width: 0.9rem;
    height: 0.9rem;
    margin-top: -0.2rem;
    margin-right: 0.5rem;
    vertical-align: middle;
    fill: $colour-white;
  }
  &.large-icon svg {
    width: 1.2rem;
    height: 1.2rem;
    margin-top: -0.2rem;
    margin-right: 0.5rem;
  }
  &.secondary {
    color: $colour-sea-blue;
    border: 0.1rem solid $colour-sea-blue;
    font-weight: 700;
    svg {
      fill: $colour-sea-blue;
    }
  }
  &.tertiary {
    background: none;
    color: $colour-dark-imperial-blue;
    font-weight: $global-weight-bold;
    padding: 0;
    text-align: left;
    &.disabled:hover {
      color: $colour-azureish-white;
      background-color: $colour-white;
    }
    svg {
      fill: $colour-dark-imperial-blue;
    }
    &.tertiary-icon__active {
      svg {
        fill: $colour-azureish-white;
      }
    }
  }
}

.button-group {
  margin-bottom: $global-margin/2;
}
