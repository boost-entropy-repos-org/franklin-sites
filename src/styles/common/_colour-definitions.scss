@import '../colours';

// PRIMARY
.colour-sapphire-blue {
  background: $colour-sapphire-blue;
  color: $colour-sky-white;
}
.colour-sea-blue {
  background: $colour-sea-blue;
  color: $colour-sky-white;
}
.colour-vivid-cerulean {
  background: $colour-vivid-cerulean;
  color: $colour-sky-white;
}
.colour-medium-turquoise {
  background: $colour-medium-turquoise;
  color: $colour-yankees-blue;
}
.colour-diamond {
  background: $colour-diamond;
  color: $colour-yankees-blue;
}
.colour-bubbles {
  background: $colour-bubbles;
  color: $colour-yankees-blue;
}
// LINKS
.colour-dark-imperial-blue {
  background: $colour-dark-imperial-blue;
}
.colour-imperial-blue {
  background: $colour-imperial-blue;
}

// GREY
.colour-yankees-blue {
  background: $colour-yankees-blue;
  color: $colour-sky-white;
}
.colour-charcoal {
  background: $colour-charcoal;
  color: $colour-sky-white;
}
.colour-independence {
  background: $colour-independence;
  color: $colour-sky-white;
}
.colour-weldon-blue {
  background: $colour-weldon-blue;
  color: $colour-sky-white;
}
.colour-pastel-blue {
  background: $colour-pastel-blue;
  color: $colour-yankees-blue;
}
.colour-azureish-white {
  background: $colour-azureish-white;
  color: $colour-yankees-blue;
}
.colour-sky-white {
  background: $colour-sky-white;
  color: $colour-yankees-blue;
}
// CURATION
.colour-reviewed {
  background: $colour-reviewed;
  color: $colour-yankees-blue;
}
.colour-unreviewed {
  background: $colour-unreviewed;
  color: $colour-yankees-blue;
}
// NAMESPACE
.colour-uniref {
  background: $colour-uniref;
  color: $colour-yankees-blue;
}
.colour-uniparc {
  background: $colour-uniparc;
  color: $colour-yankees-blue;
}
.colour-proteomes {
  background: $colour-proteomes;
  color: $colour-sky-white;
}
.colour-supporting-data {
  background: $colour-supporting-data;
  color: $colour-yankees-blue;
}
.colour-help {
  background: $colour-help;
  color: $colour-yankees-blue;
}
.colour-annotation-systems {
  background: $colour-annotation-systems;
  color: $colour-yankees-blue;
}

// svg fills
.svg-colour-reviewed {
  fill: $colour-reviewed;
}
.svg-colour-unreviewed {
  fill: $colour-unreviewed;
}

.box-colour {
  vertical-align: top;
  display: inline-block;
  width: 9em;
  height: 9em;
  margin-bottom: 4em;
  font-size: $small-font-size;
  &:before {
    content: attr(data-name);
    color: $colour-yankees-blue;
    margin-top: 10em;
    display: block;
    text-align: center;
  }
}
