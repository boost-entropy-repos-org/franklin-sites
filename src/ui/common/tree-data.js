const treeData = [{
    label: "Item 1",
    members: [{
      label: "Item 1 a",
    }, {
      label: "Item 1 b",
      members: [{
        label: "Item 1b A",
      }, {
        label: "Item 1b B",
      }]
    }]
  },
  {
    label: "Some Item 2"
  }
];

export default treeData;
