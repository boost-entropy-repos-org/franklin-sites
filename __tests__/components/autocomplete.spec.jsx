import React from 'react';
import renderer from 'react-test-renderer';
import Autocomplete from '../../src/components/autocomplete';
import { flattenedPaths } from '../../src/app/common/tree-data';

describe('Autocomplete component', () => {
  test('should render', () => {
    const component = renderer.create(
      <Autocomplete data={flattenedPaths} onSelect={d => d} />,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});

test('should filter options', () => {
  const options = [
    {
      label: 'Find this',
    },
    {
      label: 'Also find this',
    },
    {
      label: 'You FIND this?!',
    },
    {
      label: 'Found nothing',
    },
    {
      label: '',
    },
  ];
  const filtered = Autocomplete.filterOptions(options, 'find');
  const expected = [
    {
      label: 'Find this',
    },
    {
      label: 'Also find this',
    },
    {
      label: 'You FIND this?!',
    },
  ];
  expect(filtered).toEqual(expected);
});
