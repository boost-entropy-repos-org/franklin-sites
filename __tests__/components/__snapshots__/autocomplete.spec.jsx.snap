// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete component should render 1`] = `
<div
  className="dropdown-container"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      type="text"
      value=""
    />
  </form>
  <div
    className="autocomplete-menu"
  >
    <div
      className="dropdown-menu__panel"
    />
  </div>
</div>
`;

exports[`Autocomplete component with props: showDropwdownUpdated & clearOnSelect should render 1`] = `
<div
  className="dropdown-container"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder=""
      type="text"
      value=""
    />
  </form>
  <div
    className="autocomplete-menu"
  >
    <div
      className="dropdown-menu__panel"
    />
  </div>
</div>
`;
